styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java

```diff
@@ -108,30 +118,29 @@ public void transitionInto(RunState state) {
     }
   }
 
-  private String dockerRunnerStart(RunState state) throws IOException {
-    final WorkflowInstance workflowInstance = state.workflowInstance();
+  private RunSpec createRunSpec(RunState state) throws ResourceNotFoundException {
     final Optional<ExecutionDescription> executionDescriptionOpt = state.data().executionDescription();
 
     final ExecutionDescription executionDescription = executionDescriptionOpt.orElseThrow(
         () -> new ResourceNotFoundException("Missing execution description for "
                                             + state.workflowInstance().toKey()));
 
+    final String executionId = state.data().executionId()
+        // Backwards compatibility fallback
+        .orElseGet(ExecutionDescriptionHandler::createExecutionId);
```

Comment:
The intent here is to handle existing active WFIs that have already transitioned through `PREPARE`. As they transition through `SUBMITTING`, they will have no `executionId` in their `RunState.data()` and it will be created here by `DockerRunnerHandler` and added to `RunState.data()` through the `submitted` event. This is identical to current behavior.

New WFIs that transition through `PREPARE` will already have an `executionId` in their `RunState.data()` and that will be used.